File: facts_publish/src/repositories/ebsi_didr.py
from facts_publish.src.repositories.ebsi_base import EBSIClient


class DidrRepository(EBSIClient):

    def __init__(self):
        super().__init__("did-registry")

    def get_identifier(self, did: str):
        response = self.get(f"/identifiers/{did}")
        return response

    def build_create_identifier(self, doc_hash: str):
        response = self.post(f"/track-and-trace/{doc_hash}",
                             data={"jsonrpc": "2.0", "method": "buildDocumentTransaction", "id": 1})
        return response

File: facts/frontend/src/repositories/ebsi_didr.ts
import ebsiClient from '@/repositories/ebsi_base.ts'

export default {

    async createDidrTransaction(accessToken: string, payload: any) {
        try {
            return await ebsiClient.post('/did-registry/jsonrpc', payload,
                {
                    headers: {
                        "Content-type": "application/json",
                        "Authorization": `Bearer ${accessToken}`,
                    }
                }
            )
        }catch(error: any){
            throw new Error(error.response?.data?.detail ?? "Error while creating EBSI DIDR transaction", {cause: error})
        }
    },

    async confirmDidrTransaction(accessToken: string, signedTransaction: {}) {
        try {
            return await ebsiClient.post(`/did-registry/jsonrpc`, signedTransaction,
                {
                    headers: {
                        "Content-type": "application/json",
                        "Authorization": `Bearer ${accessToken}`,
                    }
                }
            )
        }catch(error: any){
            throw new Error(error.response?.data?.detail ?? "Error while confirming EBSI DIDR transaction", {cause: error})
        }
    }

}